incubator/hnc/api/v1alpha1/hierarchical_namespace.go

```diff
@@ -0,0 +1,67 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HNSState describes the state of a hierarchical namespace. The state could be
+// "missing", "ok" or "conflict". The definitions will be described below.
+type HNSState string
+
+const (
+	// Missing means the child namespace has not been created yet. This should be the default
+	// state when the HNS is just created.
+	Missing HNSState = "missing"
+
+	// Ok means the child namespace exists.
+	Ok HNSState = "ok"
+
+	// Conflict means a namespace of the same name already exists.
+	Conflict HNSState = "conflict"
+)
+
+// HierarchicalNamespaceStatus defines the observed state of HierarchicalNamespace.
+type HierarchicalNamespaceStatus struct {
+	// State describes the state of a hierarchical namespace.
```

Comment:
Can you please document the values here, and not as part of the constants, since this documentation becomes part of the CRD definition while the constants do not. See, eg: https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/api/v1alpha1/hierarchy_types.go#L126 (note the blank lines between list elements).

The constants themselves can just get an overall comment like this: https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/api/v1alpha1/hierarchy_types.go#L34

Reply:
sg! done.